### PR TITLE
Ft profile page setting #187419175

### PR DESCRIPTION
**Pull request by yvanddniyo**
**What does this PR do?**

This PR implements profile page setting by using generic and user specific data/info.

**Description of Task to be completed?**

Implement the feature to allow user to create a profile
Can be able to retrieve and see their profile once they log in 
Can be able to update their profile once they log in
Can not be able to update their email.
Updating full name, and country is compulsory

**How should this be manually tested?**

Clone the repository.
Checkout to the branch ft-profilePage-setting-#1874419175.
Run npm install to install dependencies.
Run npm run dev to start the development server.
Open a web browser and navigate to the /docs endpoint (e.g., http://localhost:PORT/docs).
Verify that the Swagger UI loads correctly.
First authorize by logging in any user and use the token provided
Explore the endpoint **GET on /api/v1/users/profile** and  **PATCH on /api/v1/users/profile** user should be able to retrieve his or her profile and able to update their existing data in their profile.
What are the relevant pivotal tracker stories?

[Finished #187441917]

**Screenshort**

Screenshot (6)
Screenshot (7)

### CODE REVIEW

**Comment by teerenzo** [comment]
squash your commits to be one

File: __test__/user.test.ts [hunk 1]
```diff
@@ -26,51 +45,78 @@ const userTestData = {
...
+      console.error('Error connecting to the database:', error);
     }
   }, 40000);
 
-  afterAll(async () => {
-    await User.destroy({ truncate: true });
-    await sequelize.close();
-  }, 20000);
-
-  let token: any;
```
Comment (teerenzo) [inline __test__/user.test.ts]: don't change other's code

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Review by Heisjabo — DISMISSED** [review]

**Review by Heisjabo — COMMENTED** [review]
your branch have conflicts

**Review by Heisjabo — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "dev": "nodemon index.ts",
     "build": "tsc",
     "migrate": "npx sequelize-cli db:migrate",
+    "migrate:test":"npx sequelize-cli db:migrate --env test",
```
Comment (teerenzo) [inline package.json]: what is this

File: src/controllers/userControllers.ts [hunk 3]
```diff
@@ -72,6 +74,8 @@ export const userLogin = async (req: Request, res: Response) => {
 };
 
 export const createUserController = async (req: Request, res: Response) => {
+  const { name, email, username, password, isMerchant } = req.body;
+
```
Comment (teerenzo) [inline src/controllers/userControllers.ts]: remove this isMerchant

**Review by niyobertin — APPROVED** [review]

**Review by teerenzo — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (yvanddniyo) [reply package.json]: this command run migration on testing database

File: src/controllers/userControllers.ts (hunk 3 above)
Comment (yvanddniyo) [reply src/controllers/userControllers.ts]: okay

**Review by teerenzo — APPROVED** [review]

**Pull request merged**